scripts/setup_gemdaq.sh

```diff
@@ -209,7 +291,19 @@ fi
 ####################
 if [[ $SYSTEM_INFO == *"lxplus"* ]];
 then
-    export DATA_PATH=/afs/cern.ch/work/$USER[0,1]/$USER/CMS_GEM/Data/gemdata
+    if [[ -z "$DATA_PATH" ]]; # Don't override existing value
+    then
+        export DATA_PATH=/afs/cern.ch/work/${USER:0:1}/$USER/CMS_GEM/Data/gemdata
+    else
+        # Make sure it's exported
+        export DATA_PATH
```

Comment:
How this is supposed to work?

Reply:
Bash manual:
```
export [-fn] [name[=word]] ...
export -p
    The supplied names are marked for automatic export to the environment of subsequently executed commands.
```

Reply:
But you do not assign anything to $DATA_PATH before...

Reply:
If `$DATA_PATH` is empty or not defined, set and export it. Else, keep the original value and export it.

Reply:
> Else, keep the original value and export it.

If the else statement is triggered, it has already been exported, and so I think the point @mexanick is making is that this is unnecessary.

Reply:
Remember this file is to be sourced. Manual:
```
source filename [arguments]
    Read and execute commands from filename in the current shell environment
```
The key here is that the sourced file is executed in the *current shell*. Not in another process (normal execution), or even in a subshell. And hence all variables are accessible, including non-exported ones. The following examples demonstrate how the the three possible execution modes work.

**tl;dr**: Any variable set within a sourced script is accessible in the parent shell, which is why `source` is useful in the first place.

### Sourcing

Executed in the current shell and current process:
```
bash ~$ A=A
bash ~$ export B=B
bash ~$ echo 'echo $A $B; PI=3.14; export E=2.72' >test.sh
bash ~$ source test.sh
A B
bash ~$ echo $PI $E
3.14 2.72
```

### Subshell

Executed in another shell in the same process:
```
bash ~$ A=A
bash ~$ export B=B
bash ~$ (echo $A $B; PI=3.14; export E=2.72) # () creates a subshell
A B
bash ~$ echo $PI $E

```

### Executing

Executed in another shell in another process:
```
bash ~$ A=A
bash ~$ export B=B
bash ~$ echo 'echo $A $B; PI=3.14; export E=2.72' >test.sh
bash ~$ chmod +x test.sh
bash ~$ ./test.sh # Executed in another process, reads the shebang
B
bash ~$ echo $PI $E

```

Reply:
well, this would be really rare case to make problems) but as bulletproof conception is ok.